### PR TITLE
Server-side light level system

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Adds a system for calculating how much "light" is in a certain position on the map.
is based on a pretty rudimentary estimation using pointlightcomponents
Perhaps not all that efficient; i really don't know.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

**Comment by mirrorcult** [comment]
would be nice to have a clientside overlay for this to see how good of an estimate it is

File: Content.Shared/Light/LightLevelSystem.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        return GetLightLevel(pos, xform.MapID);
+    }
+
+    [PublicAPI]
+    public float GetLightLevel(Vector2 pos, MapId map)
+    {
+        // todo: in the future, this should support light on planet maps.
+        var lightLevel = 0f;
+
+        foreach (var (light, xform) in EntityQuery<SharedPointLightComponent, TransformComponent>(true))
```
Comment (mirrorcult) [inline Content.Shared/Light/LightLevelSystem.cs:35]: surely there are enough lights on the map that doing an entitylookup is faster than a query + distance check

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by EmoGarbage404** [comment]
to the shed

**Comment by EmoGarbage404** [comment]
just going to close this considering how large the change in scope will be

**Pull request closed without merging**